### PR TITLE
Skip setup on --control for commands that don't need it

### PR DESCRIPTION
**Pull request by dannyroberts**
The assumption is that for ~commands that don't have a branch arg~ some commands (django-manage, ssh, etc.) ~the branch it's run on doesn't matter that much: that's why it doesn't use the --branch mechanism to force the user to think about what branch they're on. For such commands, then,~ pulling the latest code on the control machine and re-running the install steps
is an unnecessary waste of time.

~Commands that do have the --branch option ('ansible-playbook', 'deploy', etc.) have essentially declared that they are sensitive to the latest changes. These commands will continue to trigger --control to do a full reset on the control environment as it has been doing.~

I've tested it locally with `cchq --control staging django-manage shell` and `cchq --control staging update-users` and it appears to work as described, with the former skipping setup and running quickly, and the latter doing the full setup that we're currently used to.

Update:
This PR also introduces an optional arg `--control-setup={yes,no}` that can be used with (or instead of) `--control` to override a command's default and force setup to be run or not run explicitly.

The commands that skip setup by default are: `ssh`, `tmux`, `django-manage`, `run-shell-command`, and `pillow-topic-assignments`.

##### ENVIRONMENTS AFFECTED
None

### CODE REVIEW

**Comment by dannyroberts** [comment]
> Could this cause problems for someone running commcare-cloud with a no-init command when their account on the control machine has not yet been initialized?

I see, because previously it would always run init-ansible, and now it sometimes won't? I think the way the command was written, it already depends on the git repo being there and the init script already having been run once (that's where ~/init-ansible comes from, right?). So I don't think this worsens the situation, though I do think I want to visit the onboarding workflow in another PR soon

**Comment by dannyroberts** [comment]
@millerdev I went with your suggestion to decouple them, and have updated the description and title accordingly

**Comment by dannyroberts** [comment]
I'm going to add documentation for the specific commands that have the special default to the cli and markdown docs.

File: docs/commcare-cloud/commands/index.md [hunk 1]
```diff
@@ -39,6 +39,16 @@ update the code, and run the same command entered locally but with
...
 
+### `--control-setup {yes,no}`
+
+Implies --control, and overrides the command's skip_setup_on_control_by_default value.
+
+If set to 'yes', the latest version of the branch will be pulled and commcare-cloud will
+have all its dependencies updated before the command is run.
+If set to 'no', the command will be run on whatever checkout/install of commcare-cloud
+is already on the control machine.
+Otherwise, this defaults to 'no' if command.skip_setup_on_control_by_default is True, otherwise to 'yes'.
```
Comment (esoergel) [inline docs/commcare-cloud/commands/index.md:50]: :+1:

**Review by esoergel — APPROVED** [review]
Nice, I like the idea.  I assume you'll announce when it goes live?

**Comment by dannyroberts** [comment]
> Nice, I like the idea. I assume you'll announce when it goes live?

Yeah sure, I'll make sure word gets out

**Review by millerdev — APPROVED** [review]
The comment on `--control-setup` vs `skip_setup_on_control_by_default` seems most important, although not a blocker and could be addressed in a separate PR if preferred. Otherwise looks good.

File: docs/commcare-cloud/commands/index.md [hunk 2]
```diff
@@ -39,6 +39,16 @@ update the code, and run the same command entered locally but with
 you will have to remain connected to the the control machine
 for the entirety of the run.
 
+### `--control-setup {yes,no}`
+
+Implies --control, and overrides the command's skip_setup_on_control_by_default value.
```
[the remark this answers is not in the document]
Comment (dannyroberts) [reply docs/commcare-cloud/commands/index.md:44]: Yeah thanks. I'll go back and fix this

[the remark this answers is not in the document]
Comment (dannyroberts) [reply docs/commcare-cloud/commands/index.md]: 39620fd

**Review by millerdev — APPROVED** [review]
Thanks!

**Pull request merged**